dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -329,6 +334,79 @@ public void Test_Non_Admin_User_With_Portlet_Read_Attempt() throws DotDataExcept
+        testUTFCharsSet(688,767, impl);
+        testUTFCharsSet(880,1023, impl);
+        testUTFCharsSet(1024,1279, impl);
+        testUTFCharsSet(1280,1327, impl);
+        testUTFCharsSet(8192,8303, impl);
+        testUTFCharsSet(8352,8399, impl);
+        testUTFCharsSet(8448,8527, impl);
+        testUTFCharsSet(8592,8703, impl);
+        testUTFCharsSet(8704,8959, impl);
+        testUTFCharsSet(9472,9599, impl);
+        testUTFCharsSet(9600,9631, impl);
+        testUTFCharsSet(9632,9727, impl);
+        testUTFCharsSet(9728,9983, impl);
+        testUTFCharsSet(9984,10175, impl);
+    }
+
+    private void testUTFCharsSet(final int fromCode, final int toCode, final AppsAPIImpl impl) throws IOException{
+        final StringBuilder stringBuilder = new StringBuilder();
+        for(int i=fromCode; i<= toCode; i++) {
+            final String string = fromCharCode(i);
+            stringBuilder.append(string);
+        }
+        final String input = stringBuilder.toString();
+        Logger.info(AppsAPIImplTest.class,()->  String.format(" UTF Charset code from `%d` to `%d` `%s` ",fromCode, toCode, input));
+        final char [] chars = impl.bytesToCharArrayUTF(input.getBytes(StandardCharsets.UTF_8));
+        final byte [] bytes = impl.charsToBytesUTF(chars);
+        final String output = new String(bytes, StandardCharsets.UTF_8);
+        assertEquals(input, output);
+    }
+
+    /**
+     * https://www.w3schools.com/charsets/ref_html_utf8.asp
+     * @param codePoints char code see utf char codes.
+     * @return the utf string representation.
+     */
+    private static String fromCharCode(final int... codePoints) {
+        return new String(codePoints, 0, codePoints.length);
+    }
+
+    @Test
+    public void Test_Secret_Json_Serialization_No_String_Middle_Man() throws DotDataException {
+        final AppsAPIImpl impl = new AppsAPIImpl();
+        final AppSecrets secretsIn = new AppSecrets.Builder()
+                .withKey("TheKey")
+                .withHiddenSecret("hidden1", "I'm hidden")
+                .withSecret("non-hidden1", "I'm not hidden")
+                .withSecret("non-hidden5", RandomStringUtils.randomAlphanumeric(2337))
+                .withSecret("bool1", true)
+                .build();
+        final char[] toJsonAsChars = impl.toJsonAsChars(secretsIn);
+        final AppSecrets secretsOut = impl.readJson(toJsonAsChars);
+        assertEquals(secretsIn.getKey(), secretsOut.getKey());
+
+        assertEquals(secretsIn.getSecrets().size(), secretsOut.getSecrets().size());
+
+        final Set<Entry<String, Secret>> secretsInEntries = secretsIn.getSecrets().entrySet();
+        for (final Entry<String, Secret> entryIn : secretsInEntries) {
+            assertEquals(secretsIn.getKey(), secretsOut.getKey());
+            final Secret out = secretsOut.getSecrets().get(entryIn.getKey());
+            assertNotNull(out);
+            assertTrue(out.equals(entryIn.getValue()));//This does a deepEquals.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use assertEquals(x, y) instead of assertTrue(x.equals(y))](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)